searx/templates/oscar/preferences.html

```diff
@@ -348,7 +348,11 @@ <h3>{{ _('Engines') }}</h3>
                                     <th scope="col" class="text-right">{{ _("Allow") }}</th>
                                     {% endif %}
                                 </tr>
-                        {% for search_engine in engines_by_category[categ] %}
+                        {% for group, engines in engines_by_category[categ] | group_engines %}
+                        {% if not (group == 'powered by other sources' and loop.index == 1) %}
+                        <tr><th colspan=9>{{_(group)}}</th></tr>
+                        {% endif %}
+                        {% for search_engine in engines %}
```

Comment:
> From where can the user get the informations what secondary categories are defined ... 

I think we need a list of secondary categories documented somewhere in the preferences .. not just grouping ...

> I now updated the preference templates to also group categories by the "second category".

I am sorry to say, but this sort order (by secondary category) confused me / tooked a while I understand ... and when I look for the setting of an engine (aka module) I miss the alphabetic order ... is it only me?

![grafik](https://user-images.githubusercontent.com/554536/147405018-93e15f91-7bb8-41fb-a47f-f86dcee4e847.png)

Reply:
I think that's because the Simple theme aligns the new category headings in the center. It think it should look better when it's aligned to the left (and maybe has a different background for emphasis).

Reply:
>I think we need a list of secondary categories documented somewhere in the preferences .. not just grouping ...

I don't think that's necessary.

I updated the simple theme, I think the subcategories now look much better:

![image](https://user-images.githubusercontent.com/73739153/147425581-f5f69fe6-8139-4871-87fd-8b5f2e0fdb0d.png)